File: apis/backend/src/main/java/com/tms/api/objects/models/sprint/SprintModel.java
package com.tms.api.objects.models.sprint;

import java.util.Date;
import java.util.List;

import com.tms.api.objects.types.task.SprintType;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class SprintModel {
    private String sprintId;
    private String sprintName;
    private String projectId;
    private List<SprintTaskModel> tasks;
    private SprintType sprintType;
    private Date startDate;
    private Date endDate;
}

File: ticket-managment-system-web/src/app/core/models/sprint/sprint.model.ts
import { SprintType } from "../../enums";
import { SprintTaskModel } from "./sprint-task.model";

export class SprintModel {
    public sprintId!: String;
    public sprintName!: String;
    public projectId!: String;
    public tasks!: SprintTaskModel[];
    public sprintType!: SprintType;
    public startDate!: Date;
    public endDate!: Date;

    constructor(sprint: Partial<SprintModel> = {}) {
        Object.assign(this, sprint);
    }
}